diff --git a/charmap.txt b/charmap.txt
index b9d0ed9..5cf6c5a 100644
--- a/charmap.txt
+++ b/charmap.txt
@@ -910,3 +910,5 @@ EMOJI_BIGANGER = F9 FE
 '\l' = FA @ scroll up window text
 '\p' = FB @ new paragraph
 '\n' = FE @ new line
+
+RTL = FC 19
diff --git a/data/text/new_game_intro.inc b/data/text/new_game_intro.inc
index e667b68..3ac79d3 100644
--- a/data/text/new_game_intro.inc
+++ b/data/text/new_game_intro.inc
@@ -1,20 +1,20 @@
 gOtherText_NewName::
-    .string "NEW NAME$"
+    .string "{RTL}ÁÑÙ ぉÎこÎ$"
 
 gNameChoice_Green::
     .string "GREEN$"
 
 gNameChoice_Red::
-    .string "RED$"
+    .string "ÎÌïÒ$"
 
 gNameChoice_Leaf::
-    .string "LEAF$"
+    .string "+ÓÇÂ$"
 
 gNameChoice_Fire::
-    .string "FIRE$"
+    .string "ÙこÈŒ$"
 
 gNameChoice_Gary::
-    .string "GARY$"
+    .string "ÈÌò$"
 
 gNameChoice_Kaz::
     .string "KAZ$"
@@ -23,221 +23,172 @@ gNameChoice_Toru::
     .string "TORU$"
 
 gNameChoice_Ash::
-    .string "ASH$"
+    .string "ぃÀこだ$"
 
 gNameChoice_Kene::
-    .string "KENE$"
+    .string "ùÉÀÑ$"
 
 gNameChoice_Geki::
     .string "GEKI$"
 
 gNameChoice_Jak::
-    .string "JAK$"
+    .string "ケÚÀト$"
 
 gNameChoice_Janne::
     .string "JANNE$"
 
 gNameChoice_Jonn::
-    .string "JONN$"
+    .string "ÎÂÀぁ$"
 
 gNameChoice_Kamon::
     .string "KAMON$"
 
 gNameChoice_Karl::
-    .string "KARL$"
+    .string "カÚÀè$"
 
 gNameChoice_Taylor::
     .string "TAYLOR$"
 
 gNameChoice_Oscar::
-    .string "OSCAR$"
+    .string "ゃÑÊこ$"
 
 gNameChoice_Hiro::
     .string "HIRO$"
 
 gNameChoice_Max::
-    .string "MAX$"
+    .string "áだÀÉ$"
 
 gNameChoice_Jon::
-    .string "JON$"
+    .string "チìÒ$"
 
 gNameChoice_Ralph::
-    .string "RALPH$"
+    .string "ùÉÁÚ$"
 
 gNameChoice_Kay::
     .string "KAY$"
 
 gNameChoice_Tosh::
-    .string "TOSH$"
+    .string "îÑÀÔ$"
 
 gNameChoice_Roak::
-    .string "ROAK$"
+    .string "ÚぃÀì$"
 
 gNameChoice_Omi::
     .string "OMI$"
 
 gNameChoice_Jodi::
-    .string "JODI$"
+    .string "っÎま$"
 
 gNameChoice_Amanda::
-    .string "AMANDA$"
+    .string "ÛêÇÉÒ$"
 
 gNameChoice_Hillary::
-    .string "HILLARY$"
+    .string "+ÌÓÑ$"
 
 gNameChoice_Makey::
     .string "MAKEY$"
 
 gNameChoice_Michi::
-    .string "MICHI$"
+    .string "+êÉ$"
 
 gNameChoice_Paula::
-    .string "PAULA$"
+    .string "ÛìÁぃ$"
 
 gNameChoice_June::
-    .string "JUNE$"
+    .string "+êぉ$"
 
 gNameChoice_Cassie::
-    .string "CASSIE$"
+    .string "ÀêÇÂ$"
 
 gNameChoice_Rey::
-    .string "REY$"
+    .string "ÙこÚ$"
 
 gNameChoice_Seda::
     .string "SEDA$"
 
 gNameChoice_Kiko::
-    .string "KIKO$"
+    .string "Îªべ$"
 
 gNameChoice_Mina::
-    .string "MINA$"
+    .string "ÙこÈÉ$"
 
 gNameChoice_Norie::
     .string "NORIE$"
 
 gNameChoice_Sai::
-    .string "SAI$"
+    .string "ûÚÀÑ$"
 
 gNameChoice_Momo::
-    .string "MOMO$"
+    .string "ûÚÊì$"
 
 gNameChoice_Suzi::
-    .string "SUZI$"
+    .string "ぞêこだ$"
 
 gControlsGuide_Text_Intro::
-    .string "The various buttons will be explained in\n"
-    .string "the order of their importance.$"
+    .string "{RTL}ÑêぼÈノ Á¿だÚÁÚ ÁÂÌôËÓàÛ\nÔçßぞ ÒまÌÇËªÀ.$"
 
 gControlsGuide_Text_DPad::
-    .string "Moves the main character.\n"
-    .string "Also used to choose various data\n"
-    .string "headings.$"
+    .string "{RTL}こçÈÍ ÁÂぼô;ÇÛ ÁÂÈ%ÇßÇÛ.\nこßËôÎぅ Òこ)À ぇぁËÇÀÚ\nÉôËÓゃ ÁÂëÇÀìÀづ.$"
 
 gControlsGuide_Text_AButton::
-    .string "Used to confirm a choice, check\n"
-    .string "things, chat, and scroll text.$"
+    .string "{RTL}こßËôÎぅ ÂÓËぜŒÇÎ ねÁÂàç(\nねÁÂçÊÁÚ ねÏÌÈこÈ ÁÂê(.$"
 
 gControlsGuide_Text_BButton::
-    .string "Used to exit, cancel a choice,\n"
-    .string "and cancel a mode.$"
+    .string "{RTL}こßËôÎぅ ÂÓôÈねタ ねぶÂげÀウ\nÁぇぁËÇÀÚ ねぶÂげÀウ ÁÂÊぱぎ.$"
 
 gControlsGuide_Text_StartButton::
-    .string "Press this button to open the\n"
-    .string "MENU.$"
+    .string "{RTL}Áぱげテ まじÁ ÁÂぴÚ ÂàËど\nÁÂわÀ%ÌÛ.$"
 
 gControlsGuide_Text_SelectButton::
-    .string "Used to shift items and to use\n"
-    .string "a registered item.$"
+    .string "{RTL}こßËôÎぅ Âêわî Á¿ぃねÁづ\nねぇÑËôÎÁぅ ÒぃÁû ÉßざÓÛ.$"
 
 gControlsGuide_Text_LRButtons::
-    .string "If you need help playing the\n"
-    .string "game, or on how to do things,\n"
-    .string "press the L or R Button.$"
+    .string "{RTL}ぶá ÁïËざで ÉßÀòÎû èù ÁÂÓœぞ\nÒね èù ŒÇàÇÛ èœî Á¿べÇÀウび\nèÀぱげテ まじこú ÁÂぴÚこú.$"
 
 gOakSpeech_Text_AskPlayerGender::
-    .string "Now tell me. Are you a boy?\n"
-    .string "Or are you a girl?$"
+    .string "{RTL}ÒぁëÈìù Áニá. まî Òìで èË+ぐ\nÒぅ èËÀûぐ$"
 
 gPikachuIntro_Text_Page1::
-    .string "In the world which you are about to\n"
-    .string "enter, you will embark on a grand\n"
-    .string "adventure with you as the hero.\n"
-    .string "\n"
-    .string "Speak to people and check things\n"
-    .string "wherever you go, be it towns, roads,\n"
-    .string "or caves. Gather information and\n"
-    .string "hints from every source.$"
+    .string "{RTL}èù ÁÂœÀÂÙ ÁÂじぢ Òìで òÓ+ ねべº\nぃぁÊÂóび ÑËêがÓぺ èù ÉげÀÉÈû ŒëÈっ.$"
 
 gPikachuIntro_Text_Page2::
-    .string "New paths will open to you by helping\n"
-    .string "people in need, overcoming challenges,\n"
-    .string "and solving mysteries.\n"
-    .string "\n"
-    .string "At times, you will be challenged by\n"
-    .string "others and attacked by wild creatures.\n"
-    .string "Be brave and keep pushing on.$"
+    .string "{RTL}ÑËêàËど ÒÉÀÉº ぃÚね= ぉÎこÎû\nÔÌßÀòÎû ÁÂÌçËÀぉÇú ねÏざÀねだ\nÁÂËçÎこÀづ ねïî Á¿ÂげÀだ.$"
 
 gPikachuIntro_Text_Page3::
-    .string "Through your adventure, we hope\n"
-    .string "that you will interact with all sorts\n"
-    .string "of people and achieve personal growth.\n"
-    .string "That is our biggest objective.\n"
-    .string "\n"
-    .string "Press the A Button, and let your\n"
-    .string "adventure begin!$"
+    .string "{RTL}ìÈぉÊ ぁ¡ぽ ÉげÀÉÈÏº Òá ÏËœÀÉî\nÉぎ ÁÂざÌÇぎ ねÒá Ïçわぺ\nìÌÊÁ べô;ÇÀ ŒëÇÈÁ.$"
 
 gOakSpeech_Text_WelcomeToTheWorld::
-    .string "Hello, there!\n"
-    .string "Glad to meet you!\p"
-    .string "Welcome to the world of POKéMON!\p"
-    .string "My name is OAK.\p"
-    .string "People affectionately refer to me\n"
-    .string "as the POKéMON PROFESSOR.\p$"
+    .string "{RTL}ÉÈïëÀ Ôº!\nÑœÇÎ ÔÓわÀ%º!\pÒま¡ Ôº èù òÀÂÙ ÔÊŒÇÌÊá!\pÁÑÌù ÒねÍ.\pこêÀぃこêù ÁÂêÀカ ÔÌÊぃû\nÒÑËÀぷ ÁÂëÊŒÇÌÊá.\p$"
 
 gOakSpeech_Text_ThisWorld::
-    .string "This world…$"
+    .string "{RTL}まじÁ ÁÂœÀÂÙ…$"
 
 gOakSpeech_Text_IsInhabitedFarAndWide::
-    .string "…is inhabited far and wide by\n"
-    .string "creatures called POKéMON.\p$"
+    .string "{RTL}…Ïßごêó èù Œî ÉごÀá\nÉôÓÊñÀづ ÏÎò+ ÔÊŒÇÌÊá.\p$"
 
 gOakSpeech_Text_IStudyPokemon::
-    .string "For some people, POKéMON are pets.\n"
-    .string "Others use them for battling.\p"
-    .string "As for myself…\p"
-    .string "I study POKéMON as a profession.\p$"
+    .string "{RTL}Ôœツ ÁÂêÀカ こËôじねá ÁÂëÊŒÇÌÊá\nÚèÀñÀび ねエぁÈねá こわÀÏÓÊá ÔªÀ.\pÒÉÀ ÒìÀ…\pèÎÚÁÑÛ ÁÂëÊŒÇÌÊá ÉªêËù.\p$"
 
 gOakSpeech_Text_TellMeALittleAboutYourself::
-    .string "But first, tell me a little about\n"
-    .string "yourself.\p$"
+    .string "{RTL}Âごú Òねぇび ïÎオêù ñÓÇ¡\nòú ìàßº.\p$"
 
 gOakSpeech_Text_YourNameWhatIsIt::
-    .string "Let's begin with your name.\n"
-    .string "What is it?\p$"
+    .string "{RTL}ÂêëÎÒ ÔÀÑÌº.\nÉÀ まÊぐ\p$"
 
 gOakSpeech_Text_SoYourNameIsPlayer::
-    .string "Right…\n"
-    .string "So your name is {PLAYER}.$"
+    .string "…Àêßï\n.{PLAYER} ºÌÑÁ áぷぶ$"
 
 gOakSpeech_Text_WhatWasHisName::
-    .string "This is my grandson.\p"
-    .string "He's been your rival since you both\n"
-    .string "were babies.\p"
-    .string "…Erm, what was his name now?$"
+    .string "{RTL}まじÁ ïàÇÎぢ.\pぶìó ÉêÀèßº Éêじ Òá ŒêËÌÀ\nトàÓÇú ずげÇÈこú.\p…ÉÀ ÁÑÌóぐ ÂわÎ ìßÇで.\p$"
 
 gOakSpeech_Text_YourRivalsNameWhatWasIt::
-    .string "Your rival's name, what was it now?$"
+    .string "{RTL}ÁÑÙ ÉêÀèßºび ÉÀぷÁ ŒÀáぐ$"
 
 gOakSpeech_Text_ConfirmRivalName::
-    .string "…Er, was it {RIVAL}?$"
+    .string "ぐ{RIVAL} áÀŒ îま…$"
 
 gOakSpeech_Text_RememberRivalsName::
-    .string "That's right! I remember now!\n"
-    .string "His name is {RIVAL}!\p$"
+    .string "!áニÁ づÈŒじÏ !どÇçず\n!{RIVAL} óÌÑÁ\p$"
 
 gOakSpeech_Text_LetsGo::
-    .string "{PLAYER}!\p"
-    .string "Your very own POKéMON legend is\n"
-    .string "about to unfold!\p"
-    .string "A world of dreams and adventures\n"
-    .string "with POKéMON awaits! Let's go!$"
+    .string "!{PLAYER}\páÊÌÇŒÊëÂÁ ぎÉ ºÏÚÊがÑÒ\n!ÒÎëÏ áÒ ºべね +Óò\pづÁÈÉÀげÌÂÁね ぅ¡ï¿Á úÉ ÙÂÀò\n!ÀÇま !ÍÚÀナËìÀÔ áÊÌÇŒÊëÂÁ ぎÉ\p$"
diff --git a/data/text/obtain_item.inc b/data/text/obtain_item.inc
index 8d27081..3b2e10b 100644
--- a/data/text/obtain_item.inc
+++ b/data/text/obtain_item.inc
@@ -1,19 +1,17 @@
 Text_ObtainedTheX::
-	.string "Obtained the {STR_VAR_2}!$"
+	.string "!{STR_VAR_2} +Óò でÓ;ï$"
 
 Text_BagIsFull::
-	.string "The BAG is full…$"
+	.string "{RTL}ÁÂçわÇëÛ ÉÌËÓソÛ…$"
 
 Text_PutItemAway::
-	.string "{PLAYER} put the {STR_VAR_2}\n"
-	.string "in the {STR_VAR_3}.$"
+	.string "{STR_VAR_2} ぎぱね {PLAYER}\n.{STR_VAR_3} ùè$"
 
 Text_FoundOneItem::
-	.string "{PLAYER} found one {STR_VAR_2}!$"
+	.string "!{STR_VAR_2} Îぉね {PLAYER}$"
 
 Text_TooBadBagFull::
-	.string "Too bad!\n"
-	.string "The BAG is full…$"
+	.string "{RTL}ÂフÑゃ!\nÁÂçわÇëÛ ÉÌËÓソÛ…$"
 
 Text_FoundXCoins::
 	.string "{PLAYER} found\n"
diff --git a/data/text/pc.inc b/data/text/pc.inc
index 2bf09b3..65ef22c 100644
--- a/data/text/pc.inc
+++ b/data/text/pc.inc
@@ -1,17 +1,17 @@
 Text_BootedUpPC::
-	.string "{PLAYER} booted up the PC.$"
+	.string ".=ÊÑÀçÂÁ îげべ {PLAYER}$"
 
 Text_AccessWhichPC::
-	.string "Which PC should be accessed?$"
+	.string "{RTL}Òぢ ïÀÑÊ= ÏÈこÎ ÁÂÊずÊぽ ぶÂÇóぐ$"
 
 Text_AccessedSomeonesPC::
-	.string "Accessed Someone's PC.$"
+	.string "{RTL}ÏÙ èËど ïÀÑÊ= べô( ÉÀ.$"
 
 Text_OpenedPkmnStorage::
-	.string "POKéMON Storage System opened.$"
+	.string "{RTL}ÏÙ èËど ìナÀぅ Ïôぴこú ÁÂëÊŒÇÌÊá.$"
 
 Text_AccessedPlayersPC::
-	.string "Accessed {PLAYER}'s PC.$"
+	.string ".{PLAYER} =ÊÑÀï どËè ÙÏ$"
 
 Text_AccessedBillsPC::
-	.string "Accessed BILL's PC.$"
+	.string "{RTL}ÏÙ èËど ïÀÑÊ= ÔÇî.$"
diff --git a/include/characters.h b/include/characters.h
index d00ecf0..41d81f5 100644
--- a/include/characters.h
+++ b/include/characters.h
@@ -245,6 +245,7 @@
 #define EXT_CTRL_CODE_JPN                    0x15
 #define EXT_CTRL_CODE_ENG                    0x16
 #define EXT_CTRL_CODE_PAUSE_MUSIC            0x17
+#define EXT_CTRL_CODE_RTL                    0x19
 #define EXT_CTRL_CODE_RESUME_MUSIC           0x18
 
 #define TEXT_COLOR_TRANSPARENT  0x0
diff --git a/include/text.h b/include/text.h
index 7090a02..b1869ab 100644
--- a/include/text.h
+++ b/include/text.h
@@ -67,7 +67,8 @@ struct TextPrinterSubStruct
 {
     u8 glyphId:4;  // 0x14
     bool8 hasPrintBeenSpedUp:1;
-    u8 font_type_5:3;
+    u8 rtl:1;
+    u8 font_type_5:2;
     u8 downArrowDelay:5;
     u8 downArrowYPosIdx:2;
     u8 hasGlyphIdBeenSet:1;
diff --git a/src/battle_message.c b/src/battle_message.c
index a6392be..d4f6e65 100644
--- a/src/battle_message.c
+++ b/src/battle_message.c
@@ -45,65 +45,65 @@ static void ExpandBattleTextBuffPlaceholders(const u8 *src, u8 *dst);
 static const u8 sText_Empty1[] = _("");
 static const u8 sText_Trainer1LoseText[] = _("{B_TRAINER1_LOSE_TEXT}");
 static const u8 sText_Trainer2LoseText[] = _("{B_TRAINER2_LOSE_TEXT}");
-static const u8 sText_Trainer1RecallPkmn1[] = _("{B_TRAINER1_NAME}: {B_OPPONENT_MON1_NAME}, come back!");
+static const u8 sText_Trainer1RecallPkmn1[] = _("!Îò び{B_OPPONENT_MON1_NAME} :{B_TRAINER1_NAME}");
 static const u8 sText_Trainer1WinText[] = _("{B_TRAINER1_WIN_TEXT}");
-static const u8 sText_Trainer1RecallPkmn2[] = _("{B_TRAINER1_NAME}: {B_OPPONENT_MON2_NAME}, come back!");
+static const u8 sText_Trainer1RecallPkmn2[] = _("!Îò び{B_OPPONENT_MON2_NAME} :{B_TRAINER1_NAME}");
 static const u8 sText_Trainer1RecallBoth[] = _("{B_TRAINER1_NAME}: {B_OPPONENT_MON1_NAME} and\n{B_OPPONENT_MON2_NAME}, come back!");
 static const u8 sText_Trainer2WinText[] = _("{B_TRAINER2_WIN_TEXT}");
-static const u8 sText_PkmnGainedEXP[] = _("{B_BUFF1} gained{B_BUFF2}\n{B_BUFF3} EXP. Points!\p");
+static const u8 sText_PkmnGainedEXP[] = _("ぞßŒ {B_BUFF1}\n!{B_BUFF2}ûÈëぁ Ûがわì {B_BUFF3}\p");
 static const u8 sText_EmptyString4[] = _("");
-static const u8 sText_ABoosted[] = _(" a boosted");
-static const u8 sText_PkmnGrewToLv[] = _("{B_BUFF1} grew to\nLV. {B_BUFF2}!{WAIT_SE}\p");
-static const u8 sText_PkmnLearnedMove[] = _("{B_BUFF1} learned\n{B_BUFF2}!{WAIT_SE}\p");
-static const u8 sText_TryToLearnMove1[] = _("{B_BUFF1} is trying to\nlearn {B_BUFF2}.\p");
-static const u8 sText_TryToLearnMove2[] = _("But, {B_BUFF1} can't learn\nmore than four moves.\p");
-static const u8 sText_TryToLearnMove3[] = _("Delete a move to make\nroom for {B_BUFF2}?");
-static const u8 sText_PkmnForgotMove[] = _("{B_BUFF1} forgot\n{B_BUFF2}.\p");
-static const u8 sText_StopLearningMove[] = _("{PAUSE 32}Stop learning\n{B_BUFF2}?");
-static const u8 sText_DidNotLearnMove[] = _("{B_BUFF1} did not learn\n{B_BUFF2}.\p");
-static const u8 sText_UseNextPkmn[] = _("Use next POKéMON?");
-static const u8 sText_AttackMissed[] = _("{B_ATK_NAME_WITH_PREFIX}'s\nattack missed!");
-static const u8 sText_PkmnProtectedItself[] = _("{B_DEF_NAME_WITH_PREFIX}\nprotected itself!");
+static const u8 sText_ABoosted[] = _("{RTL} Éœぴだû");
+static const u8 sText_PkmnGrewToLv[] = _("+Âぶ +わÏÚÁ {B_BUFF1}\n{WAIT_SE}!{B_BUFF2} っÊËßÌÂÁ\p");
+static const u8 sText_PkmnLearnedMove[] = _("ÙÓœÏ {B_BUFF1}\n{WAIT_SE}!{B_BUFF2}\p");
+static const u8 sText_TryToLearnMove1[] = _("áÒ ぽねÀçこ {B_BUFF1}\n.{B_BUFF2} ÙÓœËこ\p");
+static const u8 sText_TryToLearnMove2[] = _("ÙÓœÏ ぎÇがËßこ ぇ {B_BUFF1} úごÂ\n.づÀŒÈï ぎÔÚÒ úÉ ÈâŒÒ\p");
+static const u8 sText_TryToLearnMove3[] = _("どßàËÂ ÛŒÈï セじçÏ îま\nぐ{B_BUFF2}ヌÂ ÀìÀごÉ");
+static const u8 sText_PkmnForgotMove[] = _("ùßì {B_BUFF1}\n.{B_BUFF2}\p");
+static const u8 sText_StopLearningMove[] = _("ÙÓœÏ úò ゃñÊËì î{PAUSE 32}ま\nぐ{B_BUFF2}");
+static const u8 sText_DidNotLearnMove[] = _("ÙÓœËこ ÙÂ {B_BUFF1}\n.{B_BUFF2}\p");
+static const u8 sText_UseNextPkmn[] = _("{RTL}まî ÏßËôÎぅ ÔÊŒÇÌÊá エぁÈぐ");
+static const u8 sText_AttackMissed[] = _("{B_ATK_NAME_WITH_PREFIX} ぅÊざま\n!セÎªÂÁ ぜがぁÒ");
+static const u8 sText_PkmnProtectedItself[] = _("{B_DEF_NAME_WITH_PREFIX}\n!óßàì +Ìï");
 static const u8 sText_AvoidedDamage[] = _("{B_DEF_NAME_WITH_PREFIX} avoided\ndamage with {B_DEF_ABILITY}!");
 static const u8 sText_PkmnMakesGroundMiss[] = _("{B_DEF_NAME_WITH_PREFIX} makes GROUND\nmoves miss with {B_DEF_ABILITY}!");
-static const u8 sText_PkmnAvoidedAttack[] = _("{B_DEF_NAME_WITH_PREFIX} avoided\nthe attack!");
-static const u8 sText_ItDoesntAffect[] = _("It doesn't affect\n{B_DEF_NAME_WITH_PREFIX}…");
-static const u8 sText_AttackerFainted[] = _("{B_ATK_NAME_WITH_PREFIX}\nfainted!\p");
-static const u8 sText_TargetFainted[] = _("{B_DEF_NAME_WITH_PREFIX}\nfainted!\p");
-static const u8 sText_PlayerGotMoney[] = _("{B_PLAYER_NAME} got ¥{B_BUFF1}\nfor winning!\p");
-static const u8 sText_PlayerWhiteout[] = _("{B_PLAYER_NAME} is out of\nusable POKéMON!\p");
+static const u8 sText_PkmnAvoidedAttack[] = _("っぃÀàÏ {B_DEF_NAME_WITH_PREFIX}\n!ぅÊざªÂÁ");
+static const u8 sText_ItDoesntAffect[] = _("+Óò ÀªÂ ÈÇオぜÏ ぇ\n…{B_DEF_NAME_WITH_PREFIX}");
+static const u8 sText_AttackerFainted[] = _("{B_ATK_NAME_WITH_PREFIX}\n!óÇÓò ùÌクÒ\p");
+static const u8 sText_TargetFainted[] = _("{B_DEF_NAME_WITH_PREFIX}\n!óÇÓò ùÌクÒ\p");
+static const u8 sText_PlayerGotMoney[] = _("¥{B_BUFF1} +Óò î;ï {B_PLAYER_NAME}\n!だÊàÂÁ ûぜèÀごÉ\p");
+static const u8 sText_PlayerWhiteout[] = _("óこÎÂ ぺëこ ÙÂ {B_PLAYER_NAME}\n!ぽÀËわÂÁ +Óò ÚぃÀñ áÊÌÇŒÊÔ\p");
 static const u8 sText_PlayerPanicked[] = _("{B_PLAYER_NAME} panicked and lost ¥{B_BUFF1}…\p… … … …\p{B_PLAYER_NAME} whited out!{PAUSE_UNTIL_PRESS}");
 static const u8 sText_PlayerWhiteoutAgainstTrainer[] = _("{B_PLAYER_NAME} is out of\nusable POKéMON!\pPlayer lost against\n{B_TRAINER1_CLASS} {B_TRAINER1_NAME}!{PAUSE_UNTIL_PRESS}");
 static const u8 sText_PlayerPaidAsPrizeMoney[] = _("{B_PLAYER_NAME} paid ¥{B_BUFF1} as the prize\nmoney…\p… … … …\p{B_PLAYER_NAME} whited out!{PAUSE_UNTIL_PRESS}");
-static const u8 sText_PlayerWhiteout2[] = _("{B_PLAYER_NAME} whited out!{PAUSE_UNTIL_PRESS}");
+static const u8 sText_PlayerWhiteout2[] = _("{PAUSE_UNTIL_PRESS}!ùòÊÂÁ Îわè {B_PLAYER_NAME}");
 static const u8 sText_PreventsEscape[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX} prevents\nescape with {B_SCR_ACTIVE_ABILITY}!\p");
-static const u8 sText_CantEscape2[] = _("Can't escape!\p");
-static const u8 sText_AttackerCantEscape[] = _("{B_ATK_NAME_WITH_PREFIX} can't escape!");
-static const u8 sText_HitXTimes[] = _("Hit {B_BUFF1} time(s)!");
-static const u8 sText_PkmnFellAsleep[] = _("{B_EFF_NAME_WITH_PREFIX}\nfell asleep!");
+static const u8 sText_CantEscape2[] = _("{RTL}ぇ こÌごú ÁÂªÈ=!\p");
+static const u8 sText_AttackerCantEscape[] = _("!=ÈªÂÁ ぎÇがËßこ ぇ {B_ATK_NAME_WITH_PREFIX}");
+static const u8 sText_HitXTimes[] = _("!ûÈÉ {B_BUFF1} óËÔÀずÒ");
+static const u8 sText_PkmnFellAsleep[] = _("{B_EFF_NAME_WITH_PREFIX}\n!ぅÊêÂÁ ùè ケÈク");
 static const u8 sText_PkmnMadeSleep[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nmade {B_EFF_NAME_WITH_PREFIX} sleep!");
 static const u8 sText_PkmnAlreadyAsleep[] = _("{B_DEF_NAME_WITH_PREFIX} is\nalready asleep!");
 static const u8 sText_PkmnAlreadyAsleep2[] = _("{B_ATK_NAME_WITH_PREFIX} is\nalready asleep!");
 static const u8 sText_PkmnWasntAffected[] = _("{B_DEF_NAME_WITH_PREFIX}\nwasn't affected!");
-static const u8 sText_PkmnWasPoisoned[] = _("{B_EFF_NAME_WITH_PREFIX}\nwas poisoned!");
+static const u8 sText_PkmnWasPoisoned[] = _("{B_EFF_NAME_WITH_PREFIX}\n!ÙÌßËÂÀÔ ぞÇずÒ");
 static const u8 sText_PkmnPoisonedBy[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\npoisoned {B_EFF_NAME_WITH_PREFIX}!");
-static const u8 sText_PkmnHurtByPoison[] = _("{B_ATK_NAME_WITH_PREFIX} is hurt\nby poison!");
+static const u8 sText_PkmnHurtByPoison[] = _("ÙÂぜËこ {B_ATK_NAME_WITH_PREFIX}\n!ÙßÂÁ úÉ");
 static const u8 sText_PkmnAlreadyPoisoned[] = _("{B_DEF_NAME_WITH_PREFIX} is already\npoisoned.");
 static const u8 sText_PkmnBadlyPoisoned[] = _("{B_EFF_NAME_WITH_PREFIX} is badly\npoisoned!");
 static const u8 sText_PkmnEnergyDrained[] = _("{B_DEF_NAME_WITH_PREFIX} had its\nenergy drained!");
-static const u8 sText_PkmnWasBurned[] = _("{B_EFF_NAME_WITH_PREFIX} was burned!");
+static const u8 sText_PkmnWasBurned[] = _("!ケÈçÔ ぞÇずÒ {B_EFF_NAME_WITH_PREFIX}");
 static const u8 sText_PkmnBurnedBy[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nburned {B_EFF_NAME_WITH_PREFIX}!");
-static const u8 sText_PkmnHurtByBurn[] = _("{B_ATK_NAME_WITH_PREFIX} is hurt\nby its burn!");
+static const u8 sText_PkmnHurtByBurn[] = _("ÙÂぜËこ {B_ATK_NAME_WITH_PREFIX}\n!óñÈï úÉ");
 static const u8 sText_PkmnAlreadyHasBurn[] = _("{B_DEF_NAME_WITH_PREFIX} already\nhas a burn.");
-static const u8 sText_PkmnWasFrozen[] = _("{B_EFF_NAME_WITH_PREFIX} was\nfrozen solid!");
+static const u8 sText_PkmnWasFrozen[] = _("ÎÌざÏ {B_EFF_NAME_WITH_PREFIX}\n!ÀÉÀÌÏ");
 static const u8 sText_PkmnFrozenBy[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nfroze {B_EFF_NAME_WITH_PREFIX} solid!");
-static const u8 sText_PkmnIsFrozen[] = _("{B_ATK_NAME_WITH_PREFIX} is\nfrozen solid!");
-static const u8 sText_PkmnWasDefrosted[] = _("{B_DEF_NAME_WITH_PREFIX} was\ndefrosted!");
+static const u8 sText_PkmnIsFrozen[] = _("ÎÌざËÉ {B_ATK_NAME_WITH_PREFIX}\n!ÀÉÀÌÏ");
+static const u8 sText_PkmnWasDefrosted[] = _("=Áぷ {B_DEF_NAME_WITH_PREFIX}\n!ÎÇÓざÂÁ óêò");
 static const u8 sText_PkmnWasDefrosted2[] = _("{B_ATK_NAME_WITH_PREFIX} was\ndefrosted!");
 static const u8 sText_PkmnWasDefrostedBy[] = _("{B_ATK_NAME_WITH_PREFIX} was\ndefrosted by {B_CURRENT_MOVE}!");
-static const u8 sText_PkmnWasParalyzed[] = _("{B_EFF_NAME_WITH_PREFIX} is paralyzed!\nIt may be unable to move!");
+static const u8 sText_PkmnWasParalyzed[] = _("!îÓぼÂÀÔ ぞÇずÒ {B_EFF_NAME_WITH_PREFIX}\n!ÛŒÈçÂÁ úò ぴざœこ Îñ");
 static const u8 sText_PkmnWasParalyzedBy[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nparalyzed {B_EFF_NAME_WITH_PREFIX}!\lIt may be unable to move!");
-static const u8 sText_PkmnIsParalyzed[] = _("{B_ATK_NAME_WITH_PREFIX} is paralyzed!\nIt can't move!");
+static const u8 sText_PkmnIsParalyzed[] = _("!ぽÊÓぼÉ {B_ATK_NAME_WITH_PREFIX}\n!ÛŒÈçÂÁ ぎÇがËßこ ぇ");
 static const u8 sText_PkmnIsAlreadyParalyzed[] = _("{B_DEF_NAME_WITH_PREFIX} is\nalready paralyzed!");
 static const u8 sText_PkmnHealedParalysis[] = _("{B_DEF_NAME_WITH_PREFIX} was\nhealed of paralysis!");
 static const u8 sText_PkmnDreamEaten[] = _("{B_DEF_NAME_WITH_PREFIX}'s\ndream was eaten!");
@@ -111,17 +111,17 @@ static const u8 sText_StatsWontIncrease[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_BUF
 static const u8 sText_StatsWontDecrease[] = _("{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}\nwon't go lower!");
 static const u8 sText_TeamStoppedWorking[] = _("Your team's {B_BUFF1}\nstopped working!");
 static const u8 sText_FoeStoppedWorking[] = _("The foe's {B_BUFF1}\nstopped working!");
-static const u8 sText_PkmnIsConfused[] = _("{B_ATK_NAME_WITH_PREFIX} is\nconfused!");
-static const u8 sText_PkmnHealedConfusion[] = _("{B_ATK_NAME_WITH_PREFIX} snapped\nout of confusion!");
-static const u8 sText_PkmnWasConfused[] = _("{B_EFF_NAME_WITH_PREFIX} became\nconfused!");
+static const u8 sText_PkmnIsConfused[] = _("{B_ATK_NAME_WITH_PREFIX}\n!ºëÏÈÉ");
+static const u8 sText_PkmnHealedConfusion[] = _("ケÀèÒ {B_ATK_NAME_WITH_PREFIX}\n!óŒÀëÏÚÁ úÉ");
+static const u8 sText_PkmnWasConfused[] = _("ぞÇずÒ {B_EFF_NAME_WITH_PREFIX}\n!ÍÀëÏÚぇÀÔ");
 static const u8 sText_PkmnAlreadyConfused[] = _("{B_DEF_NAME_WITH_PREFIX} is\nalready confused!");
-static const u8 sText_PkmnFellInLove[] = _("{B_DEF_NAME_WITH_PREFIX}\nfell in love!");
+static const u8 sText_PkmnFellInLove[] = _("{B_DEF_NAME_WITH_PREFIX}\n!ぞçÂÁ ùè ぎñね");
 static const u8 sText_PkmnInLove[] = _("{B_ATK_NAME_WITH_PREFIX} is in love\nwith {B_SCR_ACTIVE_NAME_WITH_PREFIX}!");
 static const u8 sText_PkmnImmobilizedByLove[] = _("{B_ATK_NAME_WITH_PREFIX} is\nimmobilized by love!");
 static const u8 sText_PkmnBlownAway[] = _("{B_DEF_NAME_WITH_PREFIX} was\nblown away!");
 static const u8 sText_PkmnChangedType[] = _("{B_ATK_NAME_WITH_PREFIX} transformed\ninto the {B_BUFF1} type!");
-static const u8 sText_PkmnFlinched[] = _("{B_ATK_NAME_WITH_PREFIX} flinched!");
-static const u8 sText_PkmnRegainedHealth[] = _("{B_DEF_NAME_WITH_PREFIX} regained\nhealth!");
+static const u8 sText_PkmnFlinched[] = _("!îàぉ {B_ATK_NAME_WITH_PREFIX}");
+static const u8 sText_PkmnRegainedHealth[] = _("ぃÀœËÑÁ {B_DEF_NAME_WITH_PREFIX}\n!óËÇèÀò");
 static const u8 sText_PkmnHPFull[] = _("{B_DEF_NAME_WITH_PREFIX}'s\nHP is full!");
 static const u8 sText_PkmnRaisedSpDef[] = _("{B_ATK_PREFIX2}'s {B_CURRENT_MOVE}\nraised SP. DEF!");
 static const u8 sText_PkmnRaisedSpDefALittle[] = _("{B_ATK_PREFIX2}'s {B_CURRENT_MOVE}\nraised SP. DEF a little!");
@@ -136,8 +136,8 @@ static const u8 sText_PkmnWhippedWhirlwind[] = _("{B_ATK_NAME_WITH_PREFIX} whipp
 static const u8 sText_PkmnTookSunlight[] = _("{B_ATK_NAME_WITH_PREFIX} took\nin sunlight!");
 static const u8 sText_PkmnLoweredHead[] = _("{B_ATK_NAME_WITH_PREFIX} lowered\nits head!");
 static const u8 sText_PkmnIsGlowing[] = _("{B_ATK_NAME_WITH_PREFIX} is glowing!");
-static const u8 sText_PkmnFlewHigh[] = _("{B_ATK_NAME_WITH_PREFIX} flew\nup high!");
-static const u8 sText_PkmnDugHole[] = _("{B_ATK_NAME_WITH_PREFIX} dug a hole!");
+static const u8 sText_PkmnFlewHigh[] = _("ぺÓï {B_ATK_NAME_WITH_PREFIX}\n!ÀÇÂÀò");
+static const u8 sText_PkmnDugHole[] = _("!Àわàì Èàï {B_ATK_NAME_WITH_PREFIX}");
 static const u8 sText_PkmnHidUnderwater[] = _("{B_ATK_NAME_WITH_PREFIX} hid\nunderwater!");
 static const u8 sText_PkmnSprangUp[] = _("{B_ATK_NAME_WITH_PREFIX} sprang up!");
 static const u8 sText_PkmnSqueezedByBind[] = _("{B_DEF_NAME_WITH_PREFIX} was squeezed by\n{B_ATK_NAME_WITH_PREFIX}'s BIND!");
@@ -160,7 +160,7 @@ static const u8 sText_PkmnSeeded[] = _("{B_DEF_NAME_WITH_PREFIX} was seeded!");
 static const u8 sText_PkmnEvadedAttack[] = _("{B_DEF_NAME_WITH_PREFIX} evaded\nthe attack!");
 static const u8 sText_PkmnSappedByLeechSeed[] = _("{B_ATK_NAME_WITH_PREFIX}'s health is\nsapped by LEECH SEED!");
 static const u8 sText_PkmnFastAsleep[] = _("{B_ATK_NAME_WITH_PREFIX} is fast\nasleep.");
-static const u8 sText_PkmnWokeUp[] = _("{B_ATK_NAME_WITH_PREFIX} woke up!");
+static const u8 sText_PkmnWokeUp[] = _("!ばわÇËÑÁ {B_ATK_NAME_WITH_PREFIX}");
 static const u8 sText_PkmnUproarKeptAwake[] = _("But {B_SCR_ACTIVE_NAME_WITH_PREFIX}'s UPROAR\nkept it awake!");
 static const u8 sText_PkmnWokeUpInUproar[] = _("{B_ATK_NAME_WITH_PREFIX} woke up\nin the UPROAR!");
 static const u8 sText_PkmnCausedUproar[] = _("{B_ATK_NAME_WITH_PREFIX} caused\nan UPROAR!");
@@ -211,7 +211,7 @@ static const u8 sText_PkmnCopiedStatChanges[] = _("{B_ATK_NAME_WITH_PREFIX} copi
 static const u8 sText_PkmnGotFree[] = _("{B_ATK_NAME_WITH_PREFIX} got free of\n{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}!");
 static const u8 sText_PkmnShedLeechSeed[] = _("{B_ATK_NAME_WITH_PREFIX} shed\nLEECH SEED!");
 static const u8 sText_PkmnBlewAwaySpikes[] = _("{B_ATK_NAME_WITH_PREFIX} blew away\nSPIKES!");
-static const u8 sText_PkmnFledFromBattle[] = _("{B_ATK_NAME_WITH_PREFIX} fled from\nbattle!");
+static const u8 sText_PkmnFledFromBattle[] = _("úÉ =Èま {B_ATK_NAME_WITH_PREFIX}\n!ÛŒÈœÌÂÁ");
 static const u8 sText_PkmnForesawAttack[] = _("{B_ATK_NAME_WITH_PREFIX} foresaw\nan attack!");
 static const u8 sText_PkmnTookAttack[] = _("{B_DEF_NAME_WITH_PREFIX} took the\n{B_BUFF1} attack!");
 static const u8 sText_PkmnChoseXAsDestiny[] = _("{B_ATK_NAME_WITH_PREFIX} chose\n{B_CURRENT_MOVE} as its destiny!");
@@ -299,28 +299,28 @@ static const u8 sText_PkmnsXCuredItsYProblem[] = _("{B_SCR_ACTIVE_NAME_WITH_PREF
 static const u8 sText_PkmnsXHadNoEffectOnY[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nhad no effect on {B_EFF_NAME_WITH_PREFIX}!");
 static const u8 sText_TooScaredToMove[] = _("{B_ATK_NAME_WITH_PREFIX} is too scared to move!");
 static const u8 sText_GetOutGetOut[] = _("GHOST: Get out…… Get out……");
-static const u8 sText_StatSharply[] = _("sharply ");
-const u8 gBattleText_Rose[] = _("rose!");
-static const u8 sText_StatHarshly[] = _("harshly ");
-static const u8 sText_StatFell[] = _("fell!");
-static const u8 sText_AttackersStatRose[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_BUFF1}\n{B_BUFF2}");
+static const u8 sText_StatSharply[] = _("{RTL}ÔぼÎû ");
+const u8 gBattleText_Rose[] = _("{RTL}ÁÚÏàぎ!");
+static const u8 sText_StatHarshly[] = _("{RTL}ÔぼÎû ");
+static const u8 sText_StatFell[] = _("{RTL}Áìôàツ!");
+static const u8 sText_AttackersStatRose[] = _("{B_ATK_NAME_WITH_PREFIX} {B_BUFF1}\n{B_BUFF2}");
 const u8 gText_DefendersStatRose[] = _("{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}\n{B_BUFF2}");
 static const u8 sText_UsingItemTheStatOfPkmnRose[] = _("Using {B_LAST_ITEM}, the {B_BUFF1}\nof {B_SCR_ACTIVE_NAME_WITH_PREFIX} {B_BUFF2}");
 static const u8 sText_AttackersStatFell[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_BUFF1}\n{B_BUFF2}");
 static const u8 sText_DefendersStatFell[] = _("{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}\n{B_BUFF2}");
 static const u8 sText_StatsWontIncrease2[] = _("{B_ATK_NAME_WITH_PREFIX}'s stats won't\ngo any higher!");
 static const u8 sText_StatsWontDecrease2[] = _("{B_DEF_NAME_WITH_PREFIX}'s stats won't\ngo any lower!");
-static const u8 sText_CriticalHit[] = _("A critical hit!");
-static const u8 sText_OneHitKO[] = _("It's a one-hit KO!");
+static const u8 sText_CriticalHit[] = _("{RTL}ぱÈÔÛ ïÀÑÌÛ!");
+static const u8 sText_OneHitKO[] = _("{RTL}ñ)Àウ ÏÀぅ Éú ぱÈÔÛ ねÁïÎû!");
 static const u8 sText_123Poof[] = _("{PAUSE 32}1, {PAUSE 15}2, and{PAUSE 15}… {PAUSE 15}… {PAUSE 15}… {PAUSE 15}{PLAY_SE SE_BALL_BOUNCE_1}Poof!\p");
 static const u8 sText_AndEllipsis[] = _("And…\p");
 static const u8 sText_HMMovesCantBeForgotten[] = _("HM moves can't be\nforgotten now.\p");
-static const u8 sText_NotVeryEffective[] = _("It's not very effective…");
-static const u8 sText_SuperEffective[] = _("It's super effective!");
-static const u8 sText_GotAwaySafely[] = _("{PLAY_SE SE_FLEE}Got away safely!\p");
+static const u8 sText_NotVeryEffective[] = _("{RTL}ÏぜオÇÈまÀ ÂÇチ ñÊこÀ…");
+static const u8 sText_SuperEffective[] = _("{RTL}ÏぜオÇÈまÀ ñÊぢ ÂÓげÀこÛ!");
+static const u8 sText_GotAwaySafely[] = _("{RTL}{PLAY_SE SE_FLEE}まÈÔで Ôß¡ぅ!\p");
 static const u8 sText_PkmnFledUsingIts[] = _("{PLAY_SE SE_FLEE}{B_ATK_NAME_WITH_PREFIX} fled\nusing its {B_LAST_ITEM}!\p");
 static const u8 sText_PkmnFledUsing[] = _("{PLAY_SE SE_FLEE}{B_ATK_NAME_WITH_PREFIX} fled\nusing {B_ATK_ABILITY}!\p");
-static const u8 sText_WildPkmnFled[] = _("{PLAY_SE SE_FLEE}Wild {B_BUFF1} fled!");
+static const u8 sText_WildPkmnFled[] = _("!=Èま ぢÈëÂÁ {B_BUFF1}{PLAY_SE SE_FLEE}");
 static const u8 sText_PlayerDefeatedLinkTrainer[] = _("Player defeated\n{B_LINK_OPPONENT1_NAME}!");
 static const u8 sText_TwoLinkTrainersDefeated[] = _("Player beat {B_LINK_OPPONENT1_NAME}\nand {B_LINK_OPPONENT2_NAME}!");
 static const u8 sText_PlayerLostAgainstLinkTrainer[] = _("Player lost against\n{B_LINK_OPPONENT1_NAME}!");
@@ -333,8 +333,8 @@ static const u8 sText_NoRunningFromTrainers[] = _("No! There's no running\nfrom
 static const u8 sText_CantEscape[] = _("Can't escape!\p");
 static const u8 sText_DontLeaveBirch[] = _(""); // Dummied
 static const u8 sText_ButNothingHappened[] = _("But nothing happened!");
-static const u8 sText_ButItFailed[] = _("But it failed!");
-static const u8 sText_ItHurtConfusion[] = _("It hurt itself in its\nconfusion!");
+static const u8 sText_ButItFailed[] = _("{RTL}ÂごêªÀ èぼÓで!");
+static const u8 sText_ItHurtConfusion[] = _("{RTL}エぷっ ìàßó èù\nÁÚÏëÀŒó!");
 static const u8 sText_MirrorMoveFailed[] = _("The MIRROR MOVE failed!");
 static const u8 sText_StartedToRain[] = _("It started to rain!");
 static const u8 sText_DownpourStarted[] = _("A downpour started!"); // corresponds to DownpourText in pokegold and pokecrystal and is used by Rain Dance in GSC
@@ -377,31 +377,31 @@ static const u8 sText_PkmnWatchingCarefully[] = _("{B_OPPONENT_MON1_NAME} is wat
 static const u8 sText_PkmnIsAngry[] = _("{B_OPPONENT_MON1_NAME} is angry!");
 static const u8 sText_PkmnIsEating[] = _("{B_OPPONENT_MON1_NAME} is eating!");
 static const u8 sText_OutOfSafariBalls[] = _("{PLAY_SE SE_DING_DONG}ANNOUNCER: You're out of\nSAFARI BALLS! Game over!\p");
-static const u8 sText_WildPkmnAppeared[] = _("Wild {B_OPPONENT_MON1_NAME} appeared!\p");
-static const u8 sText_WildPkmnAppeared2[] = _("Wild {B_OPPONENT_MON1_NAME} appeared!\p");
-static const u8 sText_WildPkmnAppearedPause[] = _("Wild {B_OPPONENT_MON1_NAME} appeared!{PAUSE 127}");
-static const u8 sText_TwoWildPkmnAppeared[] = _("Wild {B_OPPONENT_MON1_NAME} and\n{B_OPPONENT_MON2_NAME} appeared!\p");
+static const u8 sText_WildPkmnAppeared[] = _("!ぢÈÔ {B_OPPONENT_MON1_NAME} Èªキ\p");
+static const u8 sText_WildPkmnAppeared2[] = _("!ぢÈÔ {B_OPPONENT_MON1_NAME} Èªキ\p");
+static const u8 sText_WildPkmnAppearedPause[] = _("{PAUSE 127}!ぢÈÔ {B_OPPONENT_MON1_NAME} Èªキ");
+static const u8 sText_TwoWildPkmnAppeared[] = _("ね {B_OPPONENT_MON1_NAME} Èªキ\n!áÀこÈëÂÁ {B_OPPONENT_MON2_NAME}\p");
 static const u8 sText_GhostAppearedCantId[] = _("The GHOST appeared!\pDarn!\nThe GHOST can't be ID'd!\p");
 static const u8 sText_TheGhostAppeared[] = _("The GHOST appeared!\p");
 static const u8 sText_SilphScopeUnveil[] = _("SILPH SCOPE unveiled the GHOST's\nidentity!");
 static const u8 sText_TheGhostWas[] = _("The GHOST was MAROWAK!\p\n");
-static const u8 sText_Trainer1WantsToBattle[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME}\nwould like to battle!\p");
+static const u8 sText_Trainer1WantsToBattle[] = _("{B_TRAINER1_NAME} {B_TRAINER1_CLASS}\n!ÛÂだÀêÌÂÁ ÎこÈこ\p");
 static const u8 sText_LinkTrainerWantsToBattle[] = _("{B_LINK_OPPONENT1_NAME}\nwants to battle!");
 static const u8 sText_TwoLinkTrainersWantToBattle[] = _("{B_LINK_OPPONENT1_NAME} and {B_LINK_OPPONENT2_NAME}\nwant to battle!");
-static const u8 sText_Trainer1SentOutPkmn[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} sent\nout {B_OPPONENT_MON1_NAME}!{PAUSE 60}");
+static const u8 sText_Trainer1SentOutPkmn[] = _("îÑÚÒ {B_TRAINER1_NAME} {B_TRAINER1_CLASS}\n{PAUSE 60}!{B_OPPONENT_MON1_NAME}");
 static const u8 sText_Trainer1SentOutTwoPkmn[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} sent\nout {B_OPPONENT_MON1_NAME} and {B_OPPONENT_MON2_NAME}!{PAUSE 60}");
-static const u8 sText_Trainer1SentOutPkmn2[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} sent\nout {B_BUFF1}!");
+static const u8 sText_Trainer1SentOutPkmn2[] = _("îÑÚÒ {B_TRAINER1_NAME} {B_TRAINER1_CLASS}\n!{B_BUFF1}");
 static const u8 sText_LinkTrainerSentOutPkmn[] = _("{B_LINK_OPPONENT1_NAME} sent out\n{B_OPPONENT_MON1_NAME}!");
 static const u8 sText_LinkTrainerSentOutTwoPkmn[] = _("{B_LINK_OPPONENT1_NAME} sent out\n{B_OPPONENT_MON1_NAME} and {B_OPPONENT_MON2_NAME}!");
 static const u8 sText_TwoLinkTrainersSentOutPkmn[] = _("{B_LINK_OPPONENT1_NAME} sent out {B_LINK_OPPONENT_MON1_NAME}!\n{B_LINK_OPPONENT2_NAME} sent out {B_LINK_OPPONENT_MON2_NAME}!");
 static const u8 sText_LinkTrainerSentOutPkmn2[] = _("{B_LINK_OPPONENT1_NAME} sent out\n{B_BUFF1}!");
 static const u8 sText_LinkTrainerMultiSentOutPkmn[] = _("{B_LINK_SCR_TRAINER_NAME} sent out\n{B_BUFF1}!");
-static const u8 sText_GoPkmn[] = _("Go! {B_PLAYER_MON1_NAME}!");
+static const u8 sText_GoPkmn[] = _("!{B_PLAYER_MON1_NAME} !ÀÇま");
 static const u8 sText_GoTwoPkmn[] = _("Go! {B_PLAYER_MON1_NAME} and\n{B_PLAYER_MON2_NAME}!");
-static const u8 sText_GoPkmn2[] = _("Go! {B_BUFF1}!");
-static const u8 sText_DoItPkmn[] = _("Do it! {B_BUFF1}!");
-static const u8 sText_GoForItPkmn[] = _("Go for it, {B_BUFF1}!");
-static const u8 sText_YourFoesWeakGetEmPkmn[] = _("Your foe's weak!\nGet 'em, {B_BUFF1}!");
+static const u8 sText_GoPkmn2[] = _("!{B_BUFF1} !ÀÇま");
+static const u8 sText_DoItPkmn[] = _("!{B_BUFF1} !ÙまÚÒ");
+static const u8 sText_GoForItPkmn[] = _("!{B_BUFF1} Àこ îŒÊÏ");
+static const u8 sText_YourFoesWeakGetEmPkmn[] = _("!ゃÇœぱ ºÌ;ぁ\n!{B_BUFF1} Àこ óÇÓò ぴªぉÒ");
 static const u8 sText_LinkPartnerSentOutPkmnGoPkmn[] = _("{B_LINK_PARTNER_NAME} sent out {B_LINK_PLAYER_MON2_NAME}!\nGo! {B_LINK_PLAYER_MON1_NAME}!");
 static const u8 sText_PkmnThatsEnough[] = _("{B_BUFF1}, that's enough!\nCome back!");
 static const u8 sText_PkmnComeBack[] = _("{B_BUFF1}, come back!");
@@ -410,14 +410,14 @@ const u8 sText_PkmnGoodComeBack[] = _("{B_BUFF1}, good!\nCome back!");
 static const u8 sText_Trainer1WithdrewPkmn[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME}\nwithdrew {B_BUFF1}!");
 static const u8 sText_LinkTrainer1WithdrewPkmn[] = _("{B_LINK_OPPONENT1_NAME} withdrew\n{B_BUFF1}!");
 static const u8 sText_LinkTrainer2WithdrewPkmn[] = _("{B_LINK_SCR_TRAINER_NAME} withdrew\n{B_BUFF1}!");
-static const u8 sText_WildPkmnPrefix[] = _("Wild ");
-static const u8 sText_FoePkmnPrefix[] = _("Foe ");
-static const u8 sText_FoePkmnPrefix2[] = _("Foe");
-static const u8 sText_AllyPkmnPrefix[] = _("Ally");
-static const u8 sText_FoePkmnPrefix3[] = _("Foe");
-static const u8 sText_AllyPkmnPrefix2[] = _("Ally");
-static const u8 sText_FoePkmnPrefix4[] = _("Foe");
-static const u8 sText_AllyPkmnPrefix3[] = _("Ally");
+static const u8 sText_WildPkmnPrefix[] = _("{RTL}ÁÂëÈぢ ");
+static const u8 sText_FoePkmnPrefix[] = _("{RTL}ÁÂô;Ù ");
+static const u8 sText_FoePkmnPrefix2[] = _("{RTL}ÁÂô;Ù");
+static const u8 sText_AllyPkmnPrefix[] = _("{RTL}ÁÂçÓÇゃ");
+static const u8 sText_FoePkmnPrefix3[] = _("{RTL}ÁÂô;Ù");
+static const u8 sText_AllyPkmnPrefix2[] = _("{RTL}ÁÂçÓÇゃ");
+static const u8 sText_FoePkmnPrefix4[] = _("{RTL}ÁÂô;Ù");
+static const u8 sText_AllyPkmnPrefix3[] = _("{RTL}ÁÂçÓÇゃ");
 static const u8 sText_AttackerUsedX[] = _("{B_ATK_NAME_WITH_PREFIX} used\n{B_BUFF2}");
 static const u8 sText_ExclamationMark[] = _("!");
 static const u8 sText_ExclamationMark2[] = _("!");
@@ -468,12 +468,12 @@ static const u8 sText_DontBeAThief[] = _("Don't be a thief!");
 static const u8 sText_ItDodgedBall[] = _("It dodged the thrown BALL!\nThis POKéMON can't be caught!");
 static const u8 sText_YouMissedPkmn[] = _("You missed the POKéMON!");
 static const u8 sText_PkmnBrokeFree[] = _("Oh, no!\nThe POKéMON broke free!");
-static const u8 sText_ItAppearedCaught[] = _("Aww!\nIt appeared to be caught!");
+static const u8 sText_ItAppearedCaught[] = _("{RTL}エヒ!\nÔÎÁ ねŒぜìó ÒÉßº!");
 static const u8 sText_AarghAlmostHadIt[] = _("Aargh!\nAlmost had it!");
 static const u8 sText_ShootSoClose[] = _("Shoot!\nIt was so close, too!");
 static const u8 sText_ItDodgedBall2[] = _("よけられた!\nこいつは つかまりそうにないぞ!"); // Unused version of the Marowak ghost dodging text
-static const u8 sText_GotchaPkmnCaught[] = _("Gotcha!\n{B_OPPONENT_MON1_NAME} was caught!{WAIT_SE}{PLAY_BGM MUS_CAUGHT}\p");
-static const u8 sText_GotchaPkmnCaught2[] = _("Gotcha!\n{B_OPPONENT_MON1_NAME} was caught!{WAIT_SE}{PLAY_BGM MUS_CAUGHT}{PAUSE 127}");
+static const u8 sText_GotchaPkmnCaught[] = _("!óËごßÉÒ\n{WAIT_SE}{PLAY_BGM MUS_CAUGHT}!{B_OPPONENT_MON1_NAME} ぃÀÇがずÁ ÙÏ\p");
+static const u8 sText_GotchaPkmnCaught2[] = _("!óËごßÉÒ\n{WAIT_SE}{PLAY_BGM MUS_CAUGHT}{PAUSE 127}!{B_OPPONENT_MON1_NAME} ぃÀÇがずÁ ÙÏ");
 static const u8 sText_GiveNicknameCaptured[] = _("Give a nickname to the\ncaptured {B_OPPONENT_MON1_NAME}?");
 static const u8 sText_PkmnSentToPC[] = _("{B_OPPONENT_MON1_NAME} was sent to\n{B_PC_CREATOR_NAME} PC.");
 static const u8 sText_Someones[] = _("someone's");
@@ -1369,7 +1369,7 @@ const u8 *const gTrainerTowerChallengeTypeTexts[NUM_TOWER_CHALLENGE_TYPES] =
     gOtherText_Mixed
 };
 
-static const u8 sText_Trainer1Fled[] = _("{PLAY_SE SE_FLEE}{B_TRAINER1_CLASS} {B_TRAINER1_NAME} fled!");
+static const u8 sText_Trainer1Fled[] = _("!=Èま {B_TRAINER1_NAME} {B_TRAINER1_CLASS}{PLAY_SE SE_FLEE}");
 static const u8 sText_PlayerLostAgainstTrainer1[] = _("Player lost against\n{B_TRAINER1_CLASS} {B_TRAINER1_NAME}!");
 static const u8 sText_PlayerBattledToDrawTrainer1[] = _("Player battled to a draw against\n{B_TRAINER1_CLASS} {B_TRAINER1_NAME}!");
 
diff --git a/src/string_util.c b/src/string_util.c
index 5c26d15..439c4ff 100644
--- a/src/string_util.c
+++ b/src/string_util.c
@@ -628,6 +628,7 @@ u8 GetExtCtrlCodeLength(u8 code)
         1,
         1,
         1,
+        1,
     };
 
     u8 length = 0;
diff --git a/src/strings.c b/src/strings.c
index c099555..fa8d1f9 100644
--- a/src/strings.c
+++ b/src/strings.c
@@ -15,25 +15,25 @@ const u8 gExpandedPlaceholder_Groudon[] = _("GROUDON");
 const u8 gExpandedPlaceholder_Red[] = _("RED");
 const u8 gExpandedPlaceholder_Green[] = _("GREEN");
 const u8 gText_EggNickname[] = _("EGG");
-const u8 gText_MenuPokemon[] = _("POKéMON");
-const u8 gText_NewGame[] = _("NEW GAME");
-const u8 gText_Continue[] = _("CONTINUE");
+const u8 gText_MenuPokemon[] = _("{RTL}ÔÊŒÇÌÊá");
+const u8 gText_NewGame[] = _("{RTL}ÂœëÛ ぉÎこÎû");
+const u8 gText_Continue[] = _("{RTL}ÉËÀÔœÛ");
 const u8 gTextJP_Options[] = _("せっていを かえる");
 const u8 gUnusedText_MysteryGift[] = _("MYSTERY GIFT");
-const u8 gText_MysteryGift[] = _("MYSTERY GIFT");
+const u8 gText_MysteryGift[] = _("{RTL}まÎこÛ クÀÉ)Û");
 const u8 gText_WirelessNotConnected[] = _("The Wireless Adapter is not\nconnected.");
 const u8 gText_MysteryGiftCantUse[] = _("MYSTERY GIFT can't be used while\nthe Wireless Adapter is attached.");
 const u8 gText_UpdatingSaveExternalData[] = _("がいぶデ-タにより レポ-トを こうしんします\nしばらく おまちください");
 const u8 gText_SaveFileUpdated[] = _("レポ-トが こうしんされました!");
 const u8 gText_SaveFileUpdatedSwapBackup[] = _("レポ-トが こうしんされました!\pこれいじょう\nレポ-トが かききれないので\lバックアップカ-トリッジを\lこうかんしてください!\pくわしくは\nにんてんどう サ-ビスセンタ- まで\lおといあわせ ください");
 const u8 gText_FailedUpdateSwapBackup[] = _("レポ-トの こうしんは\nしっぱいしました!\pバックアップカ-トリッジを\nこうかんしてください!\pくわしくは\nにんてんどう サ-ビスセンタ- まで\lおといあわせ ください");
-const u8 gText_SaveFileCorrupted[] = _("The save file is corrupted.\pThe previous save file will be\nloaded.");
-const u8 gText_SaveFileHasBeenDeleted[] = _("The save file has been\ndeleted...");
+const u8 gText_SaveFileCorrupted[] = _("{RTL}ÉÓゃ ÁÂçàば ÏÀÂゃ.\pÑÇËÙ ÏçÌÇî ÉÓゃ ÁÂçàば\nÁÂßÀÔぺ.");
+const u8 gText_SaveFileHasBeenDeleted[] = _("{RTL}ÏÙ ïじセ ÉÓゃ\nÁÂçàば…");
 const u8 gText_1MSubCircuitBoardNotInstalled[] = _("The 1M sub-circuit board is\nnot installed.");
 const u8 gTextJP_InternalBatteryHasRunDry[] = _("でんちぎれの ために\nとけいが うごかなくなりました\pとけいに かんけいする できごとは おきませんが\nゲ-ムを つづけて あそぶことは できます");
 const u8 gText_PlayerUnused[] = _("しゅじんこう");
 const u8 gText_PokedexUnused[] = _("ポケモンずかん");
-const u8 gText_MainMenuTime[] = _("TIME");
+const u8 gText_MainMenuTime[] = _("でñÊÂÁ");
 const u8 gText_BadgesUnused[] = _("もっているバッジ");
 const u8 gText_AButton[] = _("Aボタン");
 const u8 gText_BButton[] = _("Bボタン");
@@ -156,13 +156,13 @@ const u8 gText_SelectorArrow2[] = _("▶");
 const u8 gText_YesNo[] = _("YES\nNO");
 const u8 gText_GoBackToThePreviousMenu[] = _("Go back to the\nprevious menu.");
 const u8 gText_WhatWouldYouLikeToDo[] = _("What would you like to do?");
-const u8 gText_MenuPokedex[] = _("POKéDEX");
-const u8 gText_MenuBag[] = _("BAG");
+const u8 gText_MenuPokedex[] = _("{RTL}ÁÂàªÈカ");
+const u8 gText_MenuBag[] = _("{RTL}ÁÂçわÇëÛ");
 const u8 gText_MenuPokenav[] = _("ポケナビ");
 const u8 gText_MenuPlayer[] = _("{PLAYER}");
-const u8 gText_MenuSave[] = _("SAVE");
-const u8 gText_MenuOption[] = _("OPTION");
-const u8 gText_MenuExit[] = _("EXIT");
+const u8 gText_MenuSave[] = _("{RTL}ïàば");
+const u8 gText_MenuOption[] = _("{RTL}ÁÂôÇÀÚÁづ");
+const u8 gText_MenuExit[] = _("{RTL}ぁÈねタ");
 const u8 gText_MenuRetire[] = _("RETIRE");
 const u8 gText_MenuSafariStats[] = _("{STR_VAR_1}/{STR_VAR_2}\nBALLS  {STR_VAR_3}");
 const u8 gText_TimesStrVar1[] = _("×{STR_VAR_1}");
@@ -306,7 +306,7 @@ ALIGNED(4) const u8 gText_OtherTrainerCantAcceptPkmn[] = _("The other TRAINER ca
 ALIGNED(4) const u8 gText_CantTradeWithTrainer[] = _("You can't trade with that\nTRAINER now.");
 ALIGNED(4) const u8 gText_NotPkmnOtherTrainerWants[] = _("That isn't the type of POKéMON\nthat the other TRAINER wants.");
 ALIGNED(4) const u8 gText_ThatIsntAnEgg[] = _("That isn't an EGG.");
-const u8 gText_ChoosePokemon[] = _("Choose a POKéMON.");
+const u8 gText_ChoosePokemon[] = _("{RTL}ÁぁËÈ ÔÊŒÇÌÊá.");
 const u8 gText_MoveToWhere[] = _("Move to where?");
 const u8 gText_TeachWhichPokemon[] = _("Teach which POKéMON?");
 const u8 gText_UseOnWhichPokemon[] = _("Use on which POKéMON?");
@@ -320,7 +320,7 @@ const u8 gText_RestoreWhichMove[] = _("Restore which move?");
 const u8 gText_BoostPp[] = _("Boost PP of which?");
 const u8 gText_DoWhatWithItem[] = _("Do what with an item?");
 const u8 gText_NoPokemonForBattle[] = _("No battling this way!");
-const u8 gText_ChoosePokemon2[] = _("Choose a POKéMON.");
+const u8 gText_ChoosePokemon2[] = _("{RTL}ÁぁËÈ ÔÊŒÇÌÊá.");
 const u8 gText_NotEnoughHp[] = _("Not enough HP‥");
 const u8 gText_ThreePkmnAreNeeded[] = _("Three {PKMN} are needed.");
 const u8 gText_PokemonCantBeSame[] = _("{PKMN} can't be the same.");
@@ -411,8 +411,8 @@ const u8 gText_OK[] = _("OK");
 const u8 gOtherText_Quit[] = _("QUIT");
 const u8 gText_SawIt[] = _("みました");
 const u8 gText_NotYet[] = _("これから");
-const u8 gText_Yes[] = _("YES");
-const u8 gText_No[] = _("NO");
+const u8 gText_Yes[] = _("{RTL}ìœÙ");
+const u8 gText_No[] = _("{RTL}ぇ");
 const u8 gText_Info4[] = _("せつめい");
 const u8 gText_SingleBattle[] = _("SINGLE BATTLE");
 const u8 gText_DoubleBattle[] = _("DOUBLE BATTLE");
@@ -893,7 +893,7 @@ const u8 gText_SaveCompleted[] = _("レポ-トを かきこみました!");
 const u8 gText_SaveFailed[] = _("レポ-トの かきこみに\nしっぱい しました‥");
 const u8 gText_NoSaveFileCantSetTime[] = _("レポ-トが ないので\nじかんの せっていは できません!");
 const u8 gText_InGameClockUsable[] = _("ゲ-ムない じかんの しゅうせい システムが\nつかえる ように なりました!");
-const u8 gText_SavingDontTurnOffThePower2[] = _("SAVING…\nDON'T TURN OFF THE POWER.");
+const u8 gText_SavingDontTurnOffThePower2[] = _("{RTL}ぉÀÚ ÁÂçàば…\nぇ Ïà;î ÁÂがÀñÛ.");
 
 // Start menu option descriptions
 const u8 gStartMenuDesc_Pokedex[] = _("A device that records POKéMON secrets\nupon meeting or catching them.");
@@ -944,8 +944,8 @@ const u8 gText_PlayersXPokmonDoesNotExist[] = _("{DYNAMIC 0x00}の{DYNAMIC 0x01}
 const u8 gText_ExitingTheChat[] = _("Exiting the chat‥");
 const u8 gText_LeaderHasLeftEndingChat[] = _("The LEADER, {DYNAMIC 0x00}, has\nleft, ending the chat.");
 const u8 gText_RegisteredTextChanged_OKtoSave[] = _("The registered text has been changed.\nIs it okay to save the game?");
-const u8 gText_RegisteredTextChanged_AlreadySavedFile[] = _("There is already a saved file.\nIs it okay to overwrite it?");
-const u8 gText_RegisteredTextChanged_SavingDontTurnOff[] = _("SAVING‥\nDON'T TURN OFF THE POWER.");
+const u8 gText_RegisteredTextChanged_AlreadySavedFile[] = _("{RTL}こÊぉÎ ÉÓゃ ÉçàÊハ ÔÀÂàœî.\nまî ÏÈこÎ ÁÂごËÀÔÛ èÊñóぐ");
+const u8 gText_RegisteredTextChanged_SavingDontTurnOff[] = _("{RTL}ぉÀÚ ÁÂçàば…\nぇ Ïà;î ÁÂがÀñÛ.");
 const u8 gText_RegisteredTextChanged_SavedTheGame[] = _("{DYNAMIC 0x00} saved the game.");
 const u8 gText_IfLeaderLeavesChatWillEnd[] = _("If the LEADER leaves, the chat\nwill end. Is that okay?");
 const u8 gText_Hello[] = _("HELLO");
@@ -1000,7 +1000,7 @@ const u8 gText_FilledStorageSpace2[] = _("There's no room left now.");
 const u8 gText_CantHoldMore[] = _("You can't hold any more!");
 const u8 gText_WantToPlayAgain2[] = _("Want to play again?");
 const u8 gText_SomeoneDroppedOut2[] = _("Somebody dropped out.\nThe link will be canceled.");
-const u8 gText_SavingDontTurnOffPower[] = _("SAVING…\nDON'T TURN OFF THE POWER.");
+const u8 gText_SavingDontTurnOffPower[] = _("{RTL}ぉÀÚ ÁÂçàば…\nぇ Ïà;î ÁÂがÀñÛ.");
 const u8 gText_CommunicationStandby4[] = _("Communication standby…");
 const u8 gText_SavingDontTurnOffPower2[] = _("SAVING...\nDON'T TURN OFF THE POWER."); // Unused
 const u8 gText_BerryPickingRecords[] = _("DODRIO BERRY-PICKING RECORDS");
diff --git a/src/text.c b/src/text.c
index f3eef07..b0f057d 100644
--- a/src/text.c
+++ b/src/text.c
@@ -100,14 +100,14 @@ const u8 gKeypadIconTiles[] = INCBIN_U8("graphics/fonts/keypad_icons.4bpp");
 static const u16 sFontSmallLatinGlyphs[] = INCBIN_U16("graphics/fonts/latin_small.hwlatfont");
 static const u8 sFontSmallLatinGlyphWidths[] = 
 {
-     5,  5,  5,  5,  5,  5,  5,  5,  5,  4,  5,  4,  4,  5, 
-     5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,
-     5,  5,  4,  5,  4,  4,  5,  5,  5,  6,  5,  5,  5,  5,
-     5,  5,  8,  7,  8,  5,  5,  5,  5,  5,  8,  8,  7,  5,
-     5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,
-     5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  8,
-     8,  8,  8,  8,  8,  8,  4,  7,  5,  5,  5,  5,  5,  5,
-     5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  4,
+     5,  4,  3,  4,  4,  8,  6,  8,  4,  7,  4,  7,  4,  4,
+     4,  3,  6,  8,  8,  6,  8,  8,  6,  8,  8,  7,  6,  5,
+     4,  4,  3,  6,  8,  7,  7,  6,  7,  6,  8,  8,  6,  6,
+     8,  5,  6,  7,  8,  7,  6,  8,  6,  7,  8,  8,  8,  8,
+     8,  6,  6,  6,  7,  7,  8,  5,  8,  8,  8,  8,  8,  7,
+     8,  4,  4,  8,  8,  8,  8,  6,  8,  8,  8,  7,  7,  5,
+     6,  4,  8,  8,  7,  8,  8,  3,  8,  8,  8,  4,  8,  8,
+     8,  8,  8,  8,  8,  6,  4,  8,  8,  6,  8,  5,  5,  4,
      5,  5,  5,  5,  5,  5,  5,  5,  5,  8,  8,  8,  8,  5,
      5,  5,  5,  5,  5,  5,  5,  7,  7,  5,  5,  5,  5,  5,
      5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,  5,
@@ -136,7 +136,7 @@ static const u8 sFontSmallLatinGlyphWidths[] =
      5,  5,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
-     8,  8,  8,  8,  8,  8,  8,  5
+     8,  8,  8,  8,  8,  8,  8,  5,
 };
 static const u16 sFontSmallJapaneseGlyphs[] = INCBIN_U16("graphics/fonts/japanese_small.fwjpnfont");
 
@@ -186,14 +186,14 @@ static const u16 sFontTallJapaneseGlyphs[] = INCBIN_U16("graphics/fonts/japanese
 static const u16 sFontNormalLatinGlyphs[] = INCBIN_U16("graphics/fonts/latin_normal.fwlatfont");
 static const u8 sFontNormalLatinGlyphWidths[] =
 {
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  8,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  8,  6,  6,  6,  6,
-     6,  6,  9,  8,  8,  6,  6,  6,  6,  6, 10,  8,  5,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  8,
-     8,  8,  8,  8,  8,  4,  6,  8,  5,  5,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
+     6,  4,  3,  4,  4,  8,  6,  8,  4,  7,  4,  7,  4,  4,
+     4,  3,  6,  9,  8,  6, 10, 10,  6,  8,  8,  7,  6,  6,
+     4,  4,  3,  6,  9,  7,  7,  6,  7,  6, 11,  9,  6,  6,
+    10,  5,  6,  8, 11,  7,  6,  8,  6,  7, 13, 11, 10,  9,
+     8,  6,  6,  6,  7,  7, 10,  5, 11,  8, 10, 11, 11,  7,
+    11,  4,  4, 10, 10, 10,  8,  6, 11,  8, 10,  7,  7,  5,
+     6,  4, 15,  9,  7, 10,  9,  3, 15, 10, 13,  4,  8, 15,
+    15, 11,  9,  9,  8,  6,  4,  8, 11,  6,  8,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6, 12, 12, 12, 12,  6,
      6,  6,  6,  6,  6,  6,  8,  8,  8,  6,  6,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
@@ -222,7 +222,7 @@ static const u8 sFontNormalLatinGlyphWidths[] =
      6,  6,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
-     8,  8,  8,  8,  8,  8,  8,  6
+     8,  8,  8,  8,  8,  8,  8,  6,
 };
 static const u16 sFontNormalJapaneseGlyphs[] = INCBIN_U16("graphics/fonts/japanese_normal.fwjpnfont");
 static const u8 sFontNormalJapaneseGlyphWidths[] =
@@ -252,14 +252,14 @@ static const u8 sFontNormalJapaneseGlyphWidths[] =
 static const u16 sFontMaleLatinGlyphs[] = INCBIN_U16("graphics/fonts/latin_male.fwlatfont");
 static const u8 sFontMaleLatinGlyphWidths[] =
 {
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  8,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  8,  6,  6,  6,  6,
-     6,  6,  9,  8,  8,  6,  6,  6,  6,  6, 10,  8,  5,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  8,
-     8,  8,  8,  8,  8,  4,  6,  8,  5,  5,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
+     6,  4,  3,  4,  4,  8,  6,  8,  4,  7,  4,  7,  4,  4,
+     4,  3,  6,  9,  8,  6, 10, 10,  6,  8,  8,  7,  6,  6,
+     4,  4,  3,  6,  9,  7,  7,  6,  7,  6, 11,  9,  6,  6,
+    10,  5,  6,  8, 11,  7,  6,  8,  6,  7, 13, 11, 10,  9,
+     8,  6,  6,  6,  7,  7, 10,  5, 11,  8, 10, 11, 11,  7,
+    11,  4,  4, 10, 10, 10,  8,  6, 11,  8, 10,  7,  7,  5,
+     6,  4, 15,  9,  7, 10,  9,  3, 15, 10, 13,  4,  8, 15,
+    15, 11,  9,  9,  8,  6,  4,  8, 11,  6,  8,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6, 12, 12, 12, 12,  6,
      6,  6,  6,  6,  6,  6,  8,  8,  8,  6,  6,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
@@ -288,7 +288,7 @@ static const u8 sFontMaleLatinGlyphWidths[] =
      6,  6,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
-     8,  8,  8,  8,  8,  8,  8,  6
+     8,  8,  8,  8,  8,  8,  8,  6,
 };
 static const u16 sFontMaleJapaneseGlyphs[] = INCBIN_U16("graphics/fonts/japanese_male.fwjpnfont");
 static const u8 sFontMaleJapaneseGlyphWidths[] = 
@@ -318,14 +318,14 @@ static const u8 sFontMaleJapaneseGlyphWidths[] =
 static const u16 sFontFemaleLatinGlyphs[] = INCBIN_U16("graphics/fonts/latin_female.fwlatfont");
 static const u8 sFontFemaleLatinGlyphWidths[] =
 {
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  8,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  8,  6,  6,  6,  6,
-     6,  6,  9,  8,  8,  6,  6,  6,  6,  6, 10,  8,  5,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  8,
-     8,  8,  8,  8,  8,  4,  6,  8,  5,  5,  6,  6,  6,  6,
-     6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
+     6,  4,  3,  4,  4,  8,  6,  8,  4,  7,  4,  7,  4,  4,
+     4,  3,  6,  9,  8,  6, 10, 10,  6,  8,  8,  7,  6,  6,
+     4,  4,  3,  6,  9,  7,  7,  6,  7,  6, 11,  9,  6,  6,
+    10,  5,  6,  8, 11,  7,  6,  8,  6,  7, 13, 11, 10,  9,
+     8,  6,  6,  6,  7,  7, 10,  5, 11,  8, 10, 11, 11,  7,
+    11,  4,  4, 10, 10, 10,  8,  6, 11,  8, 10,  7,  7,  5,
+     6,  4, 15,  9,  7, 10,  9,  3, 15, 10, 13,  4,  8, 15,
+    15, 11,  9,  9,  8,  6,  4,  8, 11,  6,  8,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6, 12, 12, 12, 12,  6,
      6,  6,  6,  6,  6,  6,  8,  8,  8,  6,  6,  6,  6,  6,
      6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
@@ -354,7 +354,7 @@ static const u8 sFontFemaleLatinGlyphWidths[] =
      6,  6,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
      8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,
-     8,  8,  8,  8,  8,  8,  8,  6
+     8,  8,  8,  8,  8,  8,  8,  6,
 };
 static const u16 sFontFemaleJapaneseGlyphs[] = INCBIN_U16("graphics/fonts/japanese_female.fwjpnfont");
 static const u8 sFontFemaleJapaneseGlyphWidths[] =
@@ -484,7 +484,9 @@ void TextPrinterDrawDownArrow(struct TextPrinter *textPrinter)
             FillWindowPixelRect(
                 textPrinter->printerTemplate.windowId,
                 textPrinter->printerTemplate.bgColor << 4 | textPrinter->printerTemplate.bgColor,
-                textPrinter->printerTemplate.currentX,
+                textPrinter->subUnion.sub.rtl
+                    ? textPrinter->printerTemplate.currentX - 10
+                    : textPrinter->printerTemplate.currentX,
                 textPrinter->printerTemplate.currentY,
                 10,
                 12);
@@ -507,7 +509,9 @@ void TextPrinterDrawDownArrow(struct TextPrinter *textPrinter)
                 0,
                 0x80,
                 0x10,
-                textPrinter->printerTemplate.currentX,
+                textPrinter->subUnion.sub.rtl
+                    ? textPrinter->printerTemplate.currentX - 10
+                    : textPrinter->printerTemplate.currentX,
                 textPrinter->printerTemplate.currentY,
                 10,
                 12);
@@ -524,7 +528,9 @@ void TextPrinterClearDownArrow(struct TextPrinter *textPrinter)
     FillWindowPixelRect(
         textPrinter->printerTemplate.windowId,
         textPrinter->printerTemplate.bgColor << 4 | textPrinter->printerTemplate.bgColor,
-        textPrinter->printerTemplate.currentX,
+        textPrinter->subUnion.sub.rtl
+            ? textPrinter->printerTemplate.currentX - 10
+            : textPrinter->printerTemplate.currentX,
         textPrinter->printerTemplate.currentY,
         10,
         12);
@@ -626,6 +632,12 @@ void DrawDownArrow(u8 windowId, u16 x, u16 y, u8 bgColor, bool8 drawArrow, u8 *c
     }
 }
 
+static s32 RtlLineStartX(struct TextPrinter *textPrinter)
+{
+    return gWindows[textPrinter->printerTemplate.windowId].window.width * 8
+         - textPrinter->printerTemplate.x;
+}
+
 u16 RenderText(struct TextPrinter *textPrinter)
 {
     struct TextPrinterSubStruct *subStruct = &textPrinter->subUnion.sub;
@@ -661,7 +673,8 @@ u16 RenderText(struct TextPrinter *textPrinter)
         switch (currChar)
         {
         case CHAR_NEWLINE:
-            textPrinter->printerTemplate.currentX = textPrinter->printerTemplate.x;
+            textPrinter->printerTemplate.currentX = textPrinter->subUnion.sub.rtl
+                ? RtlLineStartX(textPrinter) : textPrinter->printerTemplate.x;
             textPrinter->printerTemplate.currentY += gFonts[textPrinter->printerTemplate.fontId].maxLetterHeight + textPrinter->printerTemplate.lineSpacing;
             return RENDER_REPEAT;
         case PLACEHOLDER_BEGIN:
@@ -737,6 +750,10 @@ u16 RenderText(struct TextPrinter *textPrinter)
                 textPrinter->printerTemplate.currentChar++;
                 currChar = *textPrinter->printerTemplate.currentChar;
                 break;
+            case EXT_CTRL_CODE_RTL:
+                subStruct->rtl = TRUE;
+                textPrinter->printerTemplate.currentX = RtlLineStartX(textPrinter);
+                return RENDER_REPEAT;
             case EXT_CTRL_CODE_SHIFT_RIGHT:
                 textPrinter->printerTemplate.currentX = textPrinter->printerTemplate.x + *textPrinter->printerTemplate.currentChar;
                 textPrinter->printerTemplate.currentChar++;
@@ -836,6 +853,13 @@ u16 RenderText(struct TextPrinter *textPrinter)
             break;
         }
 
+        if (subStruct->rtl && !textPrinter->japanese)
+        {
+            textPrinter->printerTemplate.currentX -= gGlyphInfo.width;
+            CopyGlyphToWindow(textPrinter);
+            return RENDER_PRINT;
+        }
+
         CopyGlyphToWindow(textPrinter);
 
         if (textPrinter->minLetterSpacing)
@@ -864,7 +888,8 @@ u16 RenderText(struct TextPrinter *textPrinter)
         if (TextPrinterWaitWithDownArrow(textPrinter))
         {
             FillWindowPixelBuffer(textPrinter->printerTemplate.windowId, PIXEL_FILL(textPrinter->printerTemplate.bgColor));
-            textPrinter->printerTemplate.currentX = textPrinter->printerTemplate.x;
+            textPrinter->printerTemplate.currentX = textPrinter->subUnion.sub.rtl
+                ? RtlLineStartX(textPrinter) : textPrinter->printerTemplate.x;
             textPrinter->printerTemplate.currentY = textPrinter->printerTemplate.y;
             textPrinter->state = RENDER_STATE_HANDLE_CHAR;
         }
@@ -874,7 +899,8 @@ u16 RenderText(struct TextPrinter *textPrinter)
         {
             TextPrinterClearDownArrow(textPrinter);
             textPrinter->scrollDistance = gFonts[textPrinter->printerTemplate.fontId].maxLetterHeight + textPrinter->printerTemplate.lineSpacing;
-            textPrinter->printerTemplate.currentX = textPrinter->printerTemplate.x;
+            textPrinter->printerTemplate.currentX = textPrinter->subUnion.sub.rtl
+                ? RtlLineStartX(textPrinter) : textPrinter->printerTemplate.x;
             textPrinter->state = RENDER_STATE_SCROLL;
         }
         return RENDER_UPDATE;
